### PR TITLE
Fix rawBlindRSA to ensure RSA-RAW algorithm is on the signing key

### PR DESCRIPTION
**Pull request by thibmeu**
Certain platforms do not trigger an error when the algorithm name is not updated, while the setter is called.
This commit improves this behaviour by checking if the name is actually set.
If it's a silent fail, copy the key and import it with a new algorithm.

The commit also improves the test function.

### CODE REVIEW

File: test/jest.setup-file.ts [hunk 1]
```diff
@@ -23,16 +28,18 @@ async function mockSign(
...
         }
     }
-    return webcrypto.subtle.sign(algorithm, key, data);
-}
 
-if (typeof crypto === 'undefined') {
-    Object.assign(global, { crypto: webcrypto });
+    // webcrypto calls crypto, which is mocked. We need to restore the original implementation.
+    crypto.subtle.sign = parentSign;
+    const res = crypto.subtle.sign(algorithm, key, data);
```
Comment (Lekensteyn) [inline test/jest.setup-file.ts:40]: Does this have the desired behavior? sign can be overwritten before the promise is resolved.

Should you do something like this?
```
res.finally(() => {
    crypto.subtile.sign = mockSign;
});
```

File: src/blindrsa.ts [hunk 2]
```diff
@@ -188,6 +188,17 @@ export class BlindRSA {
     ): Promise<Uint8Array> {
         const algorithmName = privateKey.algorithm.name;
         privateKey.algorithm.name = BlindRSA.NATIVE_SUPPORT_NAME;
+        // only creates a copy if the private key algorithm name cannot be overwritten
+        if (privateKey.algorithm.name !== BlindRSA.NATIVE_SUPPORT_NAME) {
+            const privateKeyCopy = await crypto.subtle.exportKey('pkcs8', privateKey);
```
Comment (Lekensteyn) [inline src/blindrsa.ts:193]: Where does this key come from? It might fail if `privateKey.extractable` is false.

File: src/blindrsa.ts [hunk 3]
```diff
@@ -188,6 +188,17 @@ export class BlindRSA {
     ): Promise<Uint8Array> {
         const algorithmName = privateKey.algorithm.name;
         privateKey.algorithm.name = BlindRSA.NATIVE_SUPPORT_NAME;
+        // only creates a copy if the private key algorithm name cannot be overwritten
```
Comment (armfazh) [inline src/blindrsa.ts:191]: Additionally to pwu's comments: note that the [spec establishes](https://www.w3.org/TR/WebCryptoAPI/#dfn-CryptoKey) that the fields of a key are readonly.

```js
interface CryptoKey {
  readonly attribute [KeyType];
  readonly attribute boolean [extractable];
  readonly attribute object [algorithm];
  readonly attribute object [usages];
};
```

File: test/jest.setup-file.ts (hunk 1 above)
Comment (thibmeu) [reply test/jest.setup-file.ts:40]: updated

File: src/blindrsa.ts (hunk 2 above)
Comment (thibmeu) [reply src/blindrsa.ts:193]: it is passed by the caller. the error will propagate if the key is not extractable

**Review by armfazh — APPROVED** [review]

**Pull request merged**